pkg/api/types.go

```diff
@@ -533,3 +535,18 @@ func (o *OrchestratorProfile) IsVNETIntegrated() bool {
 func (p *Properties) HasAadProfile() bool {
 	return p.AADProfile != nil
 }
+
+// IsCustomEtcdVersion Checks if etcd version is NOT default 2.5.2
+func (o *OrchestratorProfile) IsCustomEtcdVersion() bool {
+	return "2.5.2" != o.KubernetesConfig.EtcdVersion
+}
+
+// GetAPIServerEtcdAPIVersion Used to set apiserver's etcdapi version
+func (o *OrchestratorProfile) GetAPIServerEtcdAPIVersion() string {
+	// if we are here, version has already been validated..
+	etcdversion, _ := semver.NewVersion(o.KubernetesConfig.EtcdVersion)
+	if 2 == etcdversion.Major() {
+		return "etcd2"
+	}
+	return "etcd3"
```

Comment:
You also need to confirm the kubernetes version. We still support 1.5.x which doesn't support etcd3

Reply:
Its an opt in, only. if no version specified it will default to what we have now

Reply:
We shoudl throw a validation error if they try to use 1.5 and etcd 3

Reply:
Yep. I would otherwise apiserver isn't going to start.

Reply:
Do we still want to do that even with opt-in only?